airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/migrate_to_inline_schemas/pipeline.py

```diff
@@ -0,0 +1,381 @@
+
+    title = "Migrate connector to inline schemas."  # type: ignore
+
+    def __init__(self, context: PipelineContext) -> None:
+        super().__init__(context)
+
+    async def _run(self) -> StepResult:  # type: ignore
+        connector = self.context.connector
+        connector_path = connector.code_directory
+        manifest_path = connector.manifest_path
+        python_path = connector.python_source_dir_path
+        logger = self.logger
+
+        json_streams = _parse_json_streams(python_path)
+        if len(json_streams) == 0:
+            return StepResult(step=self, status=StepStatus.SKIPPED, stderr="No JSON streams found.")
+
+        data = read_yaml(manifest_path)
+        if "streams" not in data:
+            return StepResult(step=self, status=StepStatus.SKIPPED, stderr="No manifest streams found.")
+
+        # find the explit ones and remove or udpate
+        json_loaders = _find_json_loaders(data, [])
+        for loader in json_loaders:
+            logger.info(f"     JSON loader ref: {loader.ref} -> {loader.file_path}")
+
+        _update_json_loaders(connector_path, data, json_streams, json_loaders)
+
+        # go through the declared streams and update the inline schemas
+        for stream in data["streams"]:
+            if isinstance(stream, str):
+                # see if reference
+                if stream.startswith("#"):
+                    yaml_stream = _load_reference(data, stream)
+                    if not yaml_stream:
+                        logger.info(f"    Stream reference not found: {stream}")
+                        continue
+                    if not _get_stream_name(yaml_stream):
+                        logger.info(f"    Stream reference name not found: {stream}")
+                        continue
+                else:
+                    logger.info(f"    Stream reference unknown: {stream}")
+                    continue
+            else:
+                yaml_stream = stream
+
+            stream_name = _get_stream_name(yaml_stream)
+            if not stream_name:
+                logger.info(f"    !! Stream name not found: {stream}")
+                continue
+            if yaml_stream.get("schema_loader") and yaml_stream["schema_loader"].get("type") == "InlineSchemaLoader":
+                continue
+
+            yaml_stream["schema_loader"] = {}
+            schema_loader = yaml_stream["schema_loader"]
+            _update_inline_schema(schema_loader, json_streams, stream_name)
+
+        write_yaml(data, manifest_path)
+        await format_prettier([manifest_path])
+
+        for json_stream in json_streams.values():
+            logger.info(f"     !! JSON schema not found: {json_stream.name}")
+
+        return StepResult(step=self, status=StepStatus.SUCCESS)
+
+
+@dataclass
+class JsonStream:
+    name: str
+    schema: dict
+    file_path: Path
+
+
+@dataclass
+class JsonLoaderNode:
+    ref: str
+    file_path: str
+
+
+def _has_subdirectory(directory: Path) -> bool:
```

Comment:
Potentially used in `CheckIfInlineCandidate` something, but commented out. You probably want to remove this too.